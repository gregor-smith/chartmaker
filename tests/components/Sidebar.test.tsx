test.todo('renders groups')


test.todo('renders search results when search state complete')
