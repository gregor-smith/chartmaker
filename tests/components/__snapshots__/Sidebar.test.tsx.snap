// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders groups with no search results unless complete and album present 1`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  width: 20rem;
  height: calc(100vh - (1rem * 2));
  margin-right: 1rem;
}

<div
  id="test-render-container"
>
  <aside
    class="emotion-0"
  >
    <div
      class="mock-chart-manager"
    >
      Charts: [{"name":"Test chart 1","rowsX":10,"rowsY":10,"shape":{"tag":"Top","size":40},"albums":[]},{"name":"Test chart 2","rowsX":3,"rowsY":7,"shape":{"tag":"Collage"},"albums":[]}]
      Active chart index: 1
    </div>
    <div
      class="mock-import-export-screenshot-buttons"
    >
      Screenshot state: {"loading":false,"scale":2}
    </div>
    <div
      class="mock-chart-shape-controls"
    >
      Shape: {"tag":"Collage"}
      Rows X: 3
      Rows Y: 7
    </div>
    <div
      class="mock-api-key-input"
    >
      API Key: Test api key
    </div>
    <div
      class="mock-search-box"
    >
      Search state: {"tag":"Waiting","query":"Test query"}
    </div>
  </aside>
</div>
`;

exports[`renders groups with no search results unless complete and album present 2`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  width: 20rem;
  height: calc(100vh - (1rem * 2));
  margin-right: 1rem;
}

<div
  id="test-render-container"
>
  <aside
    class="emotion-0"
  >
    <div
      class="mock-chart-manager"
    >
      Charts: [{"name":"Test chart 1","rowsX":10,"rowsY":10,"shape":{"tag":"Top","size":40},"albums":[]},{"name":"Test chart 2","rowsX":3,"rowsY":7,"shape":{"tag":"Collage"},"albums":[]}]
      Active chart index: 1
    </div>
    <div
      class="mock-import-export-screenshot-buttons"
    >
      Screenshot state: {"loading":false,"scale":2}
    </div>
    <div
      class="mock-chart-shape-controls"
    >
      Shape: {"tag":"Collage"}
      Rows X: 3
      Rows Y: 7
    </div>
    <div
      class="mock-api-key-input"
    >
      API Key: Test api key
    </div>
    <div
      class="mock-search-box"
    >
      Search state: {"tag":"Loading","query":"Test query"}
    </div>
  </aside>
</div>
`;

exports[`renders groups with no search results unless complete and album present 3`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  width: 20rem;
  height: calc(100vh - (1rem * 2));
  margin-right: 1rem;
}

<div
  id="test-render-container"
>
  <aside
    class="emotion-0"
  >
    <div
      class="mock-chart-manager"
    >
      Charts: [{"name":"Test chart 1","rowsX":10,"rowsY":10,"shape":{"tag":"Top","size":40},"albums":[]},{"name":"Test chart 2","rowsX":3,"rowsY":7,"shape":{"tag":"Collage"},"albums":[]}]
      Active chart index: 1
    </div>
    <div
      class="mock-import-export-screenshot-buttons"
    >
      Screenshot state: {"loading":false,"scale":2}
    </div>
    <div
      class="mock-chart-shape-controls"
    >
      Shape: {"tag":"Collage"}
      Rows X: 3
      Rows Y: 7
    </div>
    <div
      class="mock-api-key-input"
    >
      API Key: Test api key
    </div>
    <div
      class="mock-search-box"
    >
      Search state: {"tag":"Error","query":"Test query","message":"Test error"}
    </div>
  </aside>
</div>
`;

exports[`renders groups with no search results unless complete and album present 4`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  width: 20rem;
  height: calc(100vh - (1rem * 2));
  margin-right: 1rem;
}

<div
  id="test-render-container"
>
  <aside
    class="emotion-0"
  >
    <div
      class="mock-chart-manager"
    >
      Charts: [{"name":"Test chart 1","rowsX":10,"rowsY":10,"shape":{"tag":"Top","size":40},"albums":[]},{"name":"Test chart 2","rowsX":3,"rowsY":7,"shape":{"tag":"Collage"},"albums":[]}]
      Active chart index: 1
    </div>
    <div
      class="mock-import-export-screenshot-buttons"
    >
      Screenshot state: {"loading":false,"scale":2}
    </div>
    <div
      class="mock-chart-shape-controls"
    >
      Shape: {"tag":"Collage"}
      Rows X: 3
      Rows Y: 7
    </div>
    <div
      class="mock-api-key-input"
    >
      API Key: Test api key
    </div>
    <div
      class="mock-search-box"
    >
      Search state: {"tag":"Complete","query":"Test query","albums":[]}
    </div>
  </aside>
</div>
`;

exports[`renders groups with no search results unless complete and album present 5`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  width: 20rem;
  height: calc(100vh - (1rem * 2));
  margin-right: 1rem;
}

<div
  id="test-render-container"
>
  <aside
    class="emotion-0"
  >
    <div
      class="mock-chart-manager"
    >
      Charts: [{"name":"Test chart 1","rowsX":10,"rowsY":10,"shape":{"tag":"Top","size":40},"albums":[]},{"name":"Test chart 2","rowsX":3,"rowsY":7,"shape":{"tag":"Collage"},"albums":[]}]
      Active chart index: 1
    </div>
    <div
      class="mock-import-export-screenshot-buttons"
    >
      Screenshot state: {"loading":false,"scale":2}
    </div>
    <div
      class="mock-chart-shape-controls"
    >
      Shape: {"tag":"Collage"}
      Rows X: 3
      Rows Y: 7
    </div>
    <div
      class="mock-api-key-input"
    >
      API Key: Test api key
    </div>
    <div
      class="mock-search-box"
    >
      Search state: {"tag":"Complete","query":"Test query","albums":[{"placeholder":false,"id":0,"name":"Test album 0","url":"https://test.com/0"},{"placeholder":false,"id":1,"name":"Test album 1","url":"https://test.com/1"},{"placeholder":false,"id":2,"name":"Test album 2","url":"https://test.com/2"},{"placeholder":false,"id":3,"name":"Test album 3","url":"https://test.com/3"},{"placeholder":false,"id":4,"name":"Test album 4","url":"https://test.com/4"}]}
    </div>
    <div
      class="mock-search-results"
    >
      Albums: [{"placeholder":false,"id":0,"name":"Test album 0","url":"https://test.com/0"},{"placeholder":false,"id":1,"name":"Test album 1","url":"https://test.com/1"},{"placeholder":false,"id":2,"name":"Test album 2","url":"https://test.com/2"},{"placeholder":false,"id":3,"name":"Test album 3","url":"https://test.com/3"},{"placeholder":false,"id":4,"name":"Test album 4","url":"https://test.com/4"}]
    </div>
  </aside>
</div>
`;
